app/pb/events.proto

```diff
@@ -0,0 +1,304 @@
+syntax = "proto3";
+
+package org.couchers.api.events;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+
+service Events {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc CreateEvent(CreateEventReq) returns (Event) {
+    // Creates an event, only one instance, add more with ScheduleEvent
+  }
+
+  rpc ScheduleEvent(ScheduleEventReq) returns (Event) {
+    // Schedules a new occurence of an existing event
+  }
+
+  rpc UpdateEvent(UpdateEventReq) returns (Event) {
+    // Edit an event
+  }
+
+  rpc GetEvent(GetEventReq) returns (Event) {
+    // Gets an event
+  }
+
+  rpc ListEventOccurences(ListEventOccurencesReq) returns (ListEventOccurencesRes) {
+    // List all occurences of a recurring event
+  }
+
+  rpc ListEventAttendees(ListEventAttendeesReq) returns (ListEventAttendeesRes) {
+    // Paginated list of event attendees
+  }
+
+  rpc ListEventSubscribers(ListEventSubscribersReq) returns (ListEventSubscribersRes) {
+    // Paginated list of event subscribers
+  }
+
+  rpc ListEventOrganizers(ListEventOrganizersReq) returns (ListEventOrganizersRes) {
+    // Paginated list of event organizers
+  }
+
+  rpc TransferEvent(TransferEventReq) returns (Event) {
+    // Transfer an event you own to either a group or a community
+  }
+
+  rpc SetEventSubscription(SetEventSubscriptionReq) returns (Event) {
+    // Subscribe to get notifications about all occurences of an event
+  }
+
+  rpc SetEventAttendance(SetEventAttendanceReq) returns (Event) {
+    // RSVP/change going status for an occurence of an event
+  }
+
+  rpc InviteEventOrganizer(InviteEventOrganizerReq) returns (google.protobuf.Empty) {
+    // Invite another user to become an organizer of this event
+  }
+
+  rpc RemoveEventOrganizer(RemoveEventOrganizerReq) returns (google.protobuf.Empty) {
+    // Remove yourself as an event organizer
+  }
+
+  rpc ListMyEvents(ListMyEventsReq) returns (ListMyEventsRes) {
+    // Lists all subscribed, attending
+  }
+
+  // TODO: DownloadCalendarInvite
+}
+
+message OnlineEventInformation {
+  // videoconferencing link, etc
+  string link = 1;
+}
+
+message OfflineEventInformation {
+  // in-person events must have address + location
+  string address = 1;
+  double lat = 2;
+  double lng = 3;
+}
+
+enum AttendanceState {
+  ATTENDANCE_STATE_NOT_GOING = 0;
+  ATTENDANCE_STATE_MAYBE = 1;
+  ATTENDANCE_STATE_GOING = 2;
+}
+
+message Event {
+  int64 event_id = 1; // = occurence_id
+  // it's the next event occurence
+  bool is_next = 2;
```

Comment:
Please define "next". Is it only in my current polygon and does already started occurrences count?

Reply:
Thanks, great catch